ignite-base/App/Containers/APITestingScreen.js

```diff
@@ -0,0 +1,155 @@
+import { ScrollView, View, Text, LayoutAnimation, DeviceEventEmitter, TouchableOpacity } from 'react-native'
+import { connect } from 'react-redux'
+import { Metrics } from '../Themes'
+import FullButton from '../Components/FullButton'
+// For API
+import API from '../Services/Api'
+import FJSON from 'format-json'
+
+// Styles
+import styles from './Styles/APITestingScreenStyle'
+
+export default class APITestingScreen extends React.Component {
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      visibleHeight: Metrics.screenHeight
+    }
+
+    this.api = API.create()
+  }
+
+  static propTypes = {
+    navigator: PropTypes.object.isRequired
+  }
+
+  componentWillMount () {
+    // Using keyboardWillShow/Hide looks 1,000 times better, but doesn't work on Android
+    // TODO: Revisit this if Android begins to support - https://github.com/facebook/react-native/issues/3468
+    DeviceEventEmitter.addListener('keyboardDidShow', this.keyboardDidShow.bind(this))
+    DeviceEventEmitter.addListener('keyboardDidHide', this.keyboardDidHide.bind(this))
+
+    // Configure nav button
+    this.props.navigator.state.tapHamburger = () => {
+      this.props.navigator.leftDrawer.toggle()
+    }
+  }
+
+  componentWillUnmount () {
+    DeviceEventEmitter.removeAllListeners('keyboardDidShow')
+    DeviceEventEmitter.removeAllListeners('keyboardDidHide')
+  }
+
+  keyboardDidShow (e) {
+    // Animation types easeInEaseOut/linear/spring
+    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
+    let newSize = Metrics.screenHeight - e.endCoordinates.height
+    this.setState({
+      visibleHeight: newSize
+    })
+  }
+
+  keyboardDidHide (e) {
+    // Animation types easeInEaseOut/linear/spring
+    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
+    this.setState({
+      visibleHeight: Metrics.screenHeight
+    })
+  }
+
+  showResult (response, title = 'Response') {
+    this.refs.container.scrollTo({x: 0, y: 0, animated: true})
+    if (response.ok) {
+      // this.props.navigator.setState({title})
+      this.refs.result.setState({message: FJSON.plain(response.data), title: title})
+    } else {
+      this.refs.result.setState({message: response.problem, title: title})
+    }
+  }
+
+  tryEndpoint (apiEndpoint) {
+    const { label, endpoint, args } = apiEndpoint
+    this.api[endpoint].apply(this, args).then((result) => {
+      this.showResult(result, label || `${endpoint}(${args.join(', ')})`)
+    })
+  }
+
+  renderButton (apiEndpoint) {
+    const { label, endpoint, args } = apiEndpoint
+    return (
+      <FullButton text={label || `${endpoint}(${args.join(', ')})`} onPress={this.tryEndpoint.bind(this, apiEndpoint)} styles={{marginTop: 10}} key={`${endpoint}-${args.join('-')}`} />
+    )
+  }
+
+  renderButtons () {
+    const endpoints = [
+      { label: 'Get City (Boise)', endpoint: 'getCity', args: ['Boise'] },
+      { endpoint: 'getCity', args: ['Toronto', 'Fail'] }
```

Comment:
`label` is optional, `args` are passed in the order they are given, and over-arity args are ignored.